documentation/creating_a_script_plan.md

```diff
@@ -0,0 +1,369 @@
+Bash-like command-line arguments. Using PowerShell parameters is highly recommended to more easily
+pass arguments on the command-line and through plans to PowerShell scripts.
+
+For example, if you have a script to reboot your machine that takes a `timeout` parameter to specify how
+long to wait for the reboot, and a `shutdown_only` parameter to tell the script not to turn the
+machine back on:
+
+_reboot.ps1_
+```
+[CmdletBinding()]
+Param(
+  [Int]$timeout = 3,
+  [Boolean]$shutdown_only = $false
+)
+If (Test-Path -Path $env:SYSTEMROOT\sysnative\shutdown.exe) {
+  $executable = "$env:SYSTEMROOT\sysnative\shutdown.exe"
+}
+ElseIf (Test-Path -Path $env:SYSTEMROOT\system32\shutdown.exe) {
+  $executable = "$env:SYSTEMROOT\system32\shutdown.exe"
+}
+Else {
+  $executable = "shutdown.exe"
+}
+
+# Force a minimum timeout of 3 second to allow the response to be returned.
+If ($timeout -lt 3) {
+  $timeout = 3
+}
+
+$reboot_param = "/r"
+If ($shutdown_only) {
+  $reboot_param = "/s"
+}
+
+& $executable $reboot_param /t $timeout /d p:4:1
+```
+
+After turning the script into a plan with `bolt plan new --script` or `New-BoltPlan -Script`, add
+the parameters for your script to the plan:
+```yaml
+parameters:
+  timeout:
+    type: Optional[Integer]
+    default: 3
+    description: How long to wait for reboot before exiting
+  shutdown:
+    type: Optional[Boolean]
+    default: false
+    description: Whether to keep the machine shutdown (true) or bring it back up (false)
+  targets:
+    ...
+```
+
+Next add the argument to the [script step](writing_yaml_plans.md#script_step)
+```yaml
+steps:
+  - name: run_script
+    script: manage_docker/scripts/reboot.ps1
+    pwsh_params:
+      timeout: $timeout
+      shutdown: $shutdown
+    targets: $targets
+```
+
+And now your plan should look something like this:
+```yaml
+description: Reboot Windows machines
+parameters:
+  timeout:
+    type: Optional[Integer]
+    default: 3
+    description: How long to wait for reboot before exiting
+  shutdown:
+    type: Optional[Boolean]
+    default: false
+    description: Whether to keep the machine shutdown (true) or bring it back up (false)
+  targets:
+    type: TargetSpec
+    description: A list of targets to run actions on
+
+steps:
+  - name: run_script
+    script: manage_docker/scripts/reboot.ps1
+    pwsh_params:
+      timeout: $timeout
+      shutdown: $shutdown
+    targets: $targets
+
+return: $run_script
+```
+
+You can run the plan with:
+
+_\*nix shell command_
+```
+bolt plan run manage_docker::reboot -t <targets> timeout=30 shutdown=true
+```
+
+_PowerShell cmdlet_
+```powershell
+Invoke-BoltPlan -Name manage_docker::update_images -Targets <targets> timeout=30 shutdown=true
+```
+
+## When to write a task
+
+[Bolt tasks](tasks.md) are scripts with an optional metadata file. In general, scripts are easier to
+write and debug, and you should favor them over tasks unless you need to use a feature of the Task
+API. 
```

Comment:
```suggestion
write and debug, and you should favor them over tasks unless you need to use a task feature.
```

I'm unsure about referring to the Task API, since we don't really reference it in our docs explicitly. There's one passing mention in "Writing tasks", but nothing else. Maybe linking to the task spec?